File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3046_split-the-array/solutions/python3/solution.py
from collections import Counter
from typing import List

class Solution:
    def isPossibleToSplit(self, nums: List[int]) -> bool:
        n = len(nums) // 2
        freq = Counter(nums)
        
        # If any number appears more than n times, it is impossible to split
        if any(count > n for count in freq.values()):
            return False
        
        distinct_elements = len(freq)
        
        # We need at least n distinct elements for both halves
        if distinct_elements < n:
            return False
        
        return True

File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3046_split-the-array/solutions/java/solution.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public boolean isPossibleToSplit(int[] nums) {
        Map<Integer, Integer> freq = new HashMap<>();
        for (int num : nums) {
            freq.put(num, freq.getOrDefault(num, 0) + 1);
        }
        
        int n = nums.length / 2;
        int needTwo = 0;
        
        for (int count : freq.values()) {
            if (count == 1) {
                n--;
            } else if (count >= 2) {
                needTwo++;
                n -= 2;
            }
            if (n <= 0) break;
        }
        
        return n <= 0 && needTwo >= n;
    }
}
